contracts/staking/StakeManager.sol

```diff
@@ -63,37 +70,46 @@ contract StakeManager is Validator, IStakeManager, RootChainable, Lockable {
   //Mapping for epoch to totalStake for that epoch
   mapping (uint256 => State) public validatorState;
 
-  constructor () ERC721Full("Matic Validator", "MV") public {}
+  constructor (address _registry) ERC721Full("Matic Validator", "MV") public {
+    registry = _registry;
+  }
 
   // only staker
   modifier onlyStaker(uint256 validatorId) {
     require(ownerOf(validatorId) == msg.sender);
     _;
   }
 
-  function stake(uint256 amount, address signer) public {
-    stakeFor(msg.sender, amount, signer);
+  function stake(uint256 amount, address signer, bool isContract) external {
+    stakeFor(msg.sender, amount, signer, isContract);
   }
 
-  function stakeFor(address user, uint256 amount, address signer) public onlyWhenUnlocked {
+  function stakeFor(address user, uint256 amount, address signer, bool isContract) public onlyWhenUnlocked {
     require(currentValidatorSetSize() < validatorThreshold);
     require(balanceOf(user) == 0, "No second time staking");
     require(amount > MIN_DEPOSIT_SIZE);
     require(signerToValidator[signer] == 0);
 
     require(token.transferFrom(msg.sender, address(this), amount), "Transfer stake");
     totalStaked = totalStaked.add(amount);
+    address _contract = address(0x0);
+    if (isContract) {
+      _contract = address(new ValidatorContract(user, registry));
+    }
 
     validators[NFTCounter] = Validator({
       reward: 0,
       epoch: currentEpoch,
       amount: amount,
       activationEpoch: currentEpoch,
       deactivationEpoch: 0,
-      signer: signer
+      signer: signer,
+      contractAddress: _contract,
```

Comment:
`contractAddress: isContract ? address(new ValidatorContract(user, registry)) : address(0x0)` is more readable.

Reply:
I thought there was no `ternary operator` in solidity.